File: app/src/main/java/com/doordash/RestaurantDetailActivity.java
package com.doordash;

import android.os.Bundle;
import android.util.Log;
import android.view.MenuItem;

import androidx.annotation.Nullable;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.doordash.binding.binder.MenuBinder;
import com.doordash.binding.binder.common.CompositeItemBinder;
import com.doordash.binding.binder.common.ItemBinder;
import com.doordash.databinding.DetailOfRestaurantBinding;
import com.doordash.bean.PopularItem;
import com.doordash.bean.Store;
import com.doordash.binding.viewmodel.DishModel;
import com.doordash.binding.viewmodel.MenuModel;
import com.doordash.binding.viewmodel.RestaurantsModel;
import com.doordash.binding.viewmodel.StoreModel;

import java.util.ArrayList;
import java.util.List;

public class RestaurantDetailActivity extends AppCompatActivity {
    public String TAG="DD:RestaurantDetailActivity";
    int index=0;
    Store store;
    DetailOfRestaurantBinding detailOfRestaurantBinding;
    MenuModel menuModel=new MenuModel();
    final String key="index";
    ActionBar actionBar;
    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        index =getIntent().getIntExtra(key,-1);
        Log.i(TAG,"index: "+index);
        setContentView(R.layout.detail_of_restaurant);
        detailOfRestaurantBinding=DataBindingUtil.setContentView(this,R.layout.detail_of_restaurant);
        detailOfRestaurantBinding.setMenuModel(menuModel);
        detailOfRestaurantBinding.setLifecycleOwner(this);
        detailOfRestaurantBinding.setRestaurantView(this);
        detailOfRestaurantBinding.activityUsersRecycler.setLayoutManager(new LinearLayoutManager(this));

        init();
        actionBar= getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setTitle(store.getName());
    }


    public void init(){
        RestaurantsModel model=RestaurantsModel.getInstance();
        StoreModel storeModel= model.restaurants.get(index);
        store = storeModel.getStore();
        List<PopularItem> items = store.getMenus().get(0).getPopularItems();
        List<DishModel> dishlist=new ArrayList<>();
        DishModel dishModel;
        for(PopularItem item:items){
            dishModel =new DishModel(item);
            dishlist.add(dishModel);
        }
        menuModel.addAll(dishlist);
    }

    public ItemBinder<DishModel> itemViewBinder()
    {
        return new CompositeItemBinder<DishModel>(
                //BR.bindingdish: in item_dish
                new MenuBinder(BR.bindingdish, R.layout.item_dish)
        );
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
        }

        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/com/doordash/RestaurantDetailActivity.kt
package com.doordash

import android.os.Bundle
import android.util.Log
import android.view.MenuItem
import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.LinearLayoutManager
import com.doordash.bean.Store
import com.doordash.binding.binder.MenuBinder
import com.doordash.binding.binder.common.CompositeItemBinder
import com.doordash.binding.binder.common.ItemBinder
import com.doordash.binding.viewmodel.DishModel
import com.doordash.binding.viewmodel.MenuModel
import com.doordash.binding.viewmodel.RestaurantsModel
import com.doordash.databinding.DetailOfRestaurantBinding
import java.util.*

class RestaurantDetailActivity : AppCompatActivity() {
    var TAG = "DD:RestaurantDetailActivity"
    var index = 0
    lateinit var store: Store
    lateinit var detailOfRestaurantBinding: DetailOfRestaurantBinding
    var menuModel = MenuModel()
    val key = "index"
    lateinit var actionBar: ActionBar
    lateinit var model: RestaurantsModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        index = intent.getIntExtra(key, -1)
        Log.i(TAG, "index: $index")
        model = RestaurantsModel.instance
        if (model.size() > 0) {
            detailOfRestaurantBinding = DataBindingUtil.setContentView(this, R.layout.detail_of_restaurant)
            detailOfRestaurantBinding.menuModel = menuModel
            detailOfRestaurantBinding.lifecycleOwner = this
            detailOfRestaurantBinding.restaurantView = this
            detailOfRestaurantBinding.activityUsersRecycler.layoutManager = LinearLayoutManager(
                this
            )
            init()
            actionBar = supportActionBar!!
            actionBar.setDisplayHomeAsUpEnabled(true)
            actionBar.title = store.name
        }
    }

    fun init() {
        store = model.restaurants[index]
        if(!store.menus.isEmpty()) {
            val items = store.getMenus()[0].popularItems
            val dishlist: MutableList<DishModel> = ArrayList()
            var dishModel: DishModel
            for (item in items) {
                dishModel = DishModel(item)
                dishlist.add(dishModel)
            }
            menuModel.addAll(dishlist)
        }
    }

    fun itemViewBinder(): ItemBinder<DishModel> {
        return CompositeItemBinder( //BR.bindingdish: in item_dish
            MenuBinder(BR.bindingdish, R.layout.item_dish)
        )
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}